darshan-util/pydarshan/darshan/report.py

```diff
@@ -901,6 +940,11 @@ def mod_records(self, mod,
             None
 
         """
+
+        if dtype is not None and _warnings:
+            warn(f"The use of the dtype argument is deprecated in v3.3.2 and will be removed in v3.4.x", DeprecationWarning, stacklevel=2)
```

Comment:
I would say `v3.3.2.0` for deprecation message (the last version number is for pydarshan to bump independently of Darshan proper releases). It's hard to really say what version number we'd remove in, though, since it kind of depends on our release schedule in Darshan proper. Given that, I'm a little hesitant to commit to a strict schedule -- can we just say a future release?